scripts/gen_regex.py

```diff
@@ -0,0 +1,90 @@
+import argparse
+import unicodedata
+import chardata
+import pathlib
+
+DATA_PATH = pathlib.Path(resource_filename('wordfreq', 'data'))
+
+def _emoji_char_class():
+    """
+    Build a regex for emoji substitution.  First we create a regex character set
+    (like "[a-cv-z]") matching characters we consider emoji. The final regex
+    matches one such character followed by any number of spaces and identical
+    characters.
+    """
+    emoji_file = DATA_PATH / 'emoji.txt'
+
+    ranges = []
+    for i, c in enumerate(chardata.CHAR_CLASS_STRING):
+        # c represents the character class (3 corresponds to emoji)
+        if c == '3' and i >= 0x2600 and i != 0xfffd:
+            if ranges and i == ranges[-1][1] + 1:
+                ranges[-1][1] = i
+            else:
+                ranges.append([i, i])
+    out = '[%s]' % ''.join(chr(a) + '-' + chr(b) for a, b in ranges)
+
+    with emoji_file.open(mode='w') as file:
+        file.write(out)
+
+def _non_punct_class():
+    """
+    Builds a regex that matches anything that is not a one of the following
+    classes:
+    - P: punctuation
+    - S: symbols
+    - Z: separators
+    - C: control characters
+    This will classify symbols, including emoji, as punctuation; callers that
+    want to treat emoji separately should filter them out first.
+    """
+    non_punct_file = DATA_PATH / 'non_punct.txt
```

Comment:
...this is missing a trailing quotation mark.  Did you give the file a test run?